packages/allow-scripts/src/setup.js

```diff
@@ -65,26 +71,80 @@ function writeRcFile () {
     })
   }
 
-  configs.forEach(writeRcFileContent)
+  configs.forEach(c => writeRcFileContent(c))
 }
 
-function addPreinstallAFDependency () {
-  let cmd, cmdArgs
+function updatePackageJson(input) {
+  const p = addInstallParentDir('package.json')
+  const oldConf = JSON.parse(readFileSync(p, { encoding: 'utf-8' }))
+  const mergedConf = {
+    ...oldConf,
+    ...Object.fromEntries(
+      Object.entries(input)
+      .map(([sectionName, section]) => [
+        sectionName,
+        {
+          ...oldConf[sectionName],
+          ...section
+        }
+      ]))
+  }
 
-  if (existsSync('./.npmrc')) {
-    cmd = 'npm'
-    cmdArgs = ['install', '-d', '@lavamoat/preinstall-always-fail']
+  const output = JSON.stringify(mergedConf, undefined, 2)
+  const diff = prettyDiffJson(oldConf, mergedConf)
```

Comment:
I'm not 100% convinced this is worth taking on a new dependency. I'll see the diff anyway, it's in a repository (and if it's not, I evidently don't care).  
Agree it's nice tho.